python/ray/util/metrics.py

```diff
@@ -14,6 +15,11 @@
 
 logger = logging.getLogger(__name__)
 
+# Copied from Prometheus Python Client. While the regex is not part of the public API
+# for Prometheus, it's not expected to change
+# https://github.com/prometheus/client_python/blob/46eae7bae88f76951f7246d9f359f2dd5eeff110/prometheus_client/validation.py#L4
+METRIC_NAME_RE = re.compile(r"^[a-zA-Z_:][a-zA-Z0-9_:]*$")
```

Comment:
```suggestion
_METRIC_NAME_RE = re.compile(r"^[a-zA-Z_:][a-zA-Z0-9_:]*$")
```
to indicate it's not a public API (since this is in public API namespace)